pkg/shares/shares.go

```diff
@@ -191,6 +191,53 @@ func (s *Share) rawDataStartIndex() int {
 	return index
 }
 
+// RawDataWithReserved returns the raw share data while taking reserved bytes into account.
+func (s *Share) RawDataUsingReserved() (rawData []byte, err error) {
+	rawDataStartIndexUsingReserved, err := s.rawDataStartIndexUsingReserved()
+	if err != nil {
+		return nil, err
+	}
+
+	// This means share is the last share and does not have any transaction beginning in it
+	if rawDataStartIndexUsingReserved == 0 {
+		return []byte{}, nil
+	}
```

Comment:
I don't think so, a `tx` cannot start at zero index of a share because a properly constructed valid share has the namespace at index 0. As long as a share has a new transaction beginning in it, reserved bytes should not be 0.

Reply:
> this is the first compact share

The first compact share has reserved bytes populated.

> or the tx happens to start at zero

Not possible because the namespace version exists at index 0.

Reply:
oh shoot okay forever ago the index was relative to that index, this makes more sense thanks